### PR TITLE
Fix code smells

### PR DESCRIPTION
**Pull request by ngonzalezpazFC**
Add serialization test

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/chipfield/ChipField.java [hunk 1]
```diff
@@ -302,6 +302,7 @@ public void setNewItemHandler(SerializableFunction<String, T> handler) {
 	}
 
 	@Override
+	@SuppressWarnings("squid:S1186")
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/chipfield/ChipField.java:305]: It's not clear why the method does not have a method body: is it an unintentional omission? is it not supported? is it an intentionally-blank override? Remove the `@SuppressWarnings`

**Review by javier-godoy — CHANGES_REQUESTED** [review]
Address the other code smells mentioned in https://sonarcloud.io/project/issues?id=FlowingCode_ChipFieldAddon&resolved=false&types=CODE_SMELL

**Review by javier-godoy — CHANGES_REQUESTED** [review]
Please rebase (because this PR was based on the old master). It's not clear which of the commits belong to this PR.

**Comment by javier-godoy** [comment]
BTW 2ca18b3: `feat: implement Serializable`

**Comment by ngonzalezpazFC** [comment]
> 
> 
> Please rebase (because this PR was based on the old master). It's not clear which of the commits belong to this PR.

Should this [commit ](https://github.com/FlowingCode/ChipFieldAddon/pull/23/commits/2903eea28db3db4cd9154443b0d28dfbeeb0307c)message be "feat: implement Serializable"?

**Comment by javier-godoy** [comment]
> Should this commit (2903eea2) message be "feat: implement Serializable"?
No, because the commit only adds the serialization test (the component was already serializable)

**Review by javier-godoy — APPROVED** [review]

**Pull request merged**